ts/Series/Treegraph/TreegraphLayout.ts

```diff
@@ -0,0 +1,367 @@
+                    let gapSize = link.toNode.column - node.column - 1,
+                        dummyNode,
+                        parent = node,
+                        child = link.toNode;
+                    // parent -> dummyNode -> child
+                    let createDummyNode = function (
+                        parent: TreegraphNode,
+                        child: TreegraphNode
+                    ): TreegraphNode {
+                        // initialise dummy node.
+                        dummyNode = new TreegraphNode();
+                        dummyNode.linksFrom = [];
+                        dummyNode.linksTo = [];
+                        dummyNode.id = link.toNode.id + '-' + gapSize;
+                        dummyNode.shapeArgs = {};
+                        dummyNode.mod = 0;
+                        dummyNode.ancestor = parent;
+                        dummyNode.shift = 0;
+                        dummyNode.thread = void 0 as any;
+                        dummyNode.change = 0;
+                        dummyNode.preX = 0;
+
+                        // add connection from new node to the previous points
+
+                        // first connection to itself:
+                        dummyNode.linksFrom.push(merge(parent.linksFrom[0]));
+                        dummyNode.linksFrom[0].fromNode = dummyNode;
+                        dummyNode.linksTo.push(merge(parent.linksFrom[0]));
+                        dummyNode.linksTo[0].toNode = dummyNode;
+                        dummyNode.linksTo[0].fromNode = parent;
+                        dummyNode.linksFrom[0].toNode = child;
+                        // temporary:
+                        dummyNode.linksTo[0].to = void 0 as any;
+
+                        dummyNode.column = child.column - gapSize;
+                        dummyNode.level = child.level - gapSize;
+
+                        // Then connection from parent to dummyNode
+                        if (!parent.linksFrom[index].trueToNode) {
+                            parent.linksFrom[index].trueToNode = child;
+                        }
+                        parent.linksFrom[index].toNode = dummyNode;
+
+                        // Then connection from chhild to dummyNode
+                        if (!child.linksTo[0].trueFromNode) {
+                            child.linksTo[0].trueFromNode = parent;
+                        }
+                        child.linksTo[0].fromNode = dummyNode;
+                        return dummyNode;
+                    };
+                    while (gapSize > 0) {
+                        child = createDummyNode(parent, child);
+                        gapSize--;
+                    }
+                }
+            });
+        });
+    },
+    // Init the values of the node. probably redundant, since this can be
+    // performed in the init of the chart, but might be useful for repositioning
+    // of the nodes on the update.
```

Comment:
Have we tested if moving it to the chart init will cause problems after the update? Could it be the part of the nodes constructor?